python/lance/tests/test_schema_evolution.py

```diff
@@ -30,3 +32,34 @@ def test_write_versioned_dataset(tmp_path: Path):
     actual_df = new_dataset.to_table().to_pandas()
     expected_df = pd.DataFrame({"a": [1, 10], "b": [2, 20], "c": ["a0", "a1"]})
     pd.testing.assert_frame_equal(expected_df, actual_df)
+
+
+def test_add_column_with_literal(tmp_path: Path):
+    table = pa.Table.from_pylist([{"a": i} for i in range(10)])
+
+    base_dir = tmp_path / "test"
+    lance.write_dataset(table, base_dir)
+    dataset = lance.dataset(base_dir)
+    new_dataset = dataset.append_column(pa.field("b", pa.float64()), pc.scalar(0.5))
+
+    assert new_dataset.version["version"] == 2
+    actual_df = new_dataset.to_table().to_pandas()
+    expected_df = table.to_pandas()
+    expected_df["b"] = 0.5
+    pd.testing.assert_frame_equal(actual_df, expected_df)
+
+
+def test_add_column_with_compute(tmp_path: Path):
+    table = pa.Table.from_pylist([{"a": i} for i in range(10)])
+
+    base_dir = tmp_path / "test"
+    lance.write_dataset(table, base_dir)
+    dataset = lance.dataset(base_dir)
+    new_dataset = dataset.append_column(pa.field("b", pa.int64()),
+                                        pc.Expression._call("power", [pc.field("a"), pc.scalar(2)]))
```

Comment:
Why not call pc.power directly? https://arrow.apache.org/docs/python/generated/pyarrow.compute.power.html

Reply:
`pc.power()` is executing the function. Here we'd like to pass an expression to the runtime to execute later.